cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,202 @@
+
+
+PROD_VERSION = version.current_version().product_version()
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_create_volume(provider):
+    """Creates a volume for given cloud provider"""
+    navigate_to(Volume, 'All')
+    toolbar.select('Configuration', 'Add a new Cloud Volume')
+    fields = [('volume_name', "//input[@name='name']"),
+              ('volume_size', "//input[@name='size']"),
+              ('cloud_tenant', "//select[@id='cloud_tenant_id']")]
+    form = Form(fields=fields)
+    vname = gen_alphanumeric()
+    vsize = 1
+    volume_data = dict(volume_name=vname,
+                       volume_size=vsize,
+                       cloud_tenant=provider.mgmt.list_tenant()[0])
+    form.fill(volume_data)
+    submit_btn_loc = "//div[@id='buttons_on']/button[contains(text(), 'Add')]"
+    pytest_selenium.click(submit_btn_loc)
+    msg = pytest_selenium.text("//div[@id='flash_text_div']/div/strong")
+    assert 'Create Cloud Volume' in msg
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    pytest_selenium.refresh()
+    params = {'Name': vname,
+              'Size': '{} GB'.format(vsize),
+              'Status': 'available'}
+    res = list_tbl.find_rows_by_cells(params)
+    assert res, 'Newly created volume doesn\'t appear in UI'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_list_volumes_provider_details(provider):
+    """Verifies that all provider's volumes are present on details page"""
+    # Gather all provider's volumes name
+    volumes = []
+    for volume in provider.mgmt.list_volume():
+        vname = provider.mgmt.get_volume(volume).to_dict()['display_name']
+        volumes.append(vname)
+    provider.load_details()
+    err_msg = 'Details page shows wrong number of provider volumes'
+    clv_el = provider.summary.relationships.cloud_volumes
+    assert len(volumes) == clv_el.value, err_msg
+
+    clv_el.click()
+    err_msg = 'One of volumes is not displayed'
+    for volume in volumes:
+        assert list_tbl.find_cell('Name', volume), err_msg
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_attach_volume_from_instance_page(provider):
+    """Attaches pre-created volume to an instance from Instance page"""
+    vms = filter(lambda vm: vm.power_state == 'ACTIVE', provider.mgmt.all_vms())
+    instance_name = choice(vms).name
+    volume_name = None
+    # Find available volume
+    for volume in provider.mgmt.list_volume():
+        vol_info = provider.mgmt.get_volume(volume).to_dict()
+        if vol_info['status'] == 'available':
+            volume_name = vol_info['display_name']
+            break
+    assert volume_name, "Can't find free volume to attach"
+
+    Instance(instance_name, provider).load_details()
+    toolbar.select('Configuration', 'Attach a Cloud Volume to this Instance')
+    select = Select("//select[@id='volume_id']")
```

Comment:
Hey you can use:
list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table"
list_tbl.select_rows_by_indexes(0)
if you import from cfme.web_ui CheckboxTable